api/v1alpha1/azurevalidator_types.go

```diff
@@ -55,6 +58,31 @@ type RBACRule struct {
 	PrincipalID string `json:"principalId" yaml:"principalId"`
 }
 
+// Conveys that one or more images in a community gallery exist.
+type CommunityGalleryImageRule struct {
+	// Unique identifier for the rule in the validator. Used to ensure conditions do not overwrite
+	// each other.
+	Name string `json:"name" yaml:"name"`
+	// Gallery is the community gallery.
+	Gallery CommunityGallery `json:"gallery" yaml:"gallery"`
+	// Images are the names of the images to check for in the gallery.
+	//+kubebuilder:validation:MinItems=1
+	//+kubebuilder:validation:MaxItems=20
+	Images []string `json:"images" yaml:"images"`
+	// The subscription ID to verify that the community galleries are available to. Used in API
+	// calls to URL
+	// `subscriptions/<subscription>/providers/Microsoft.Compute/locations/<location>/communityGalleries/<gallery>/images`.
```

Comment:
Is this level of information really necessary?

Reply:
Probably not, good catch. I found that community galleries are supposed to be usable by anybody, but that in order to use them via the REST API, you have to specify them with respect to a subscription. Therefore, I felt the urge to document why subscription was in the spec. I'll come up with simpler docs here though for it.